Assets/Scripts/Interactables/DamagePopup.cs

```diff
@@ -7,58 +7,58 @@ public class DamagePopup : MonoBehaviour
     private TMP_Text label;
 
     [SerializeField]
-    private float lowDamage = 20;
+    private float scaleMultiplier = 2f;
 
     [SerializeField]
-    private float highDamage = 80;
-
-    [SerializeField]
-    private Color normalLowColor;
-
-    [SerializeField]
-    private Color normalHighColor;
-
-    [SerializeField]
-    private Color criticalLowColor;
-
-    [SerializeField]
-    private Color criticalHighColor;
+    private bool isHUD = false;
 
     public Transform Camera { get; set; }
 
+    private float lastDamage;
+
     public float Damage
     {
         set
         {
-            label.text = Mathf.RoundToInt(value).ToString();
+            label.text = isHUD 
+                ? $"-{Mathf.RoundToInt(value)}%"
+                : Mathf.RoundToInt(value).ToString();
+            lastDamage = value;
         }
         get
         {
-            return int.Parse(label.text);
+            return lastDamage;
         }
     }
 
     public bool IsCritical { get; set; }
 
     private void Start()
     {
-        label.color = DetermineColor();
+        if (IsCritical && !isHUD)
+            LeanTween.sequence()
+                .append(LeanTween.value(gameObject, (Color value) => label.color = value, ColorPallete.Health(1), ColorPallete.Health(0.2f), 0.5f).setEasePunch())
+                .append(LeanTween.value(gameObject, (float value) => label.color = ColorPallete.Health(value), 1, 0, 1f).setEaseInOutExpo());
```

Comment:
This bit is a lil confusing, but it seems you alternate between regular color interpolation and using the palette?

(and yeah this is where you'll have to change it up to avoid having the text end up orange)

Reply:
Fixed by adding gradient animation